src/solvers/refinement/string_refinement_util.cpp

```diff
@@ -322,18 +322,22 @@ std::vector<mp_integer> string_insertion_builtin_functiont::eval(
   const std::vector<mp_integer> &args_value) const
 {
   PRECONDITION(args_value.size() >= 1 || args_value.size() <= 3);
-  const std::size_t &offset = numeric_cast_v<std::size_t>(args_value[0]);
-  const std::size_t &start =
-    args_value.size() > 1 ? numeric_cast_v<std::size_t>(args_value[1]) : 0;
-  const std::size_t &end = args_value.size() > 2
-                             ? numeric_cast_v<std::size_t>(args_value[2])
-                             : input2_value.size();
+  const auto offset = std::max<BigInt::llong_t>(args_value[0].to_long(), 0);
+  const auto start = std::max<BigInt::llong_t>(
+    args_value.size() > 1 ? args_value[1].to_long() : 0, 0);
+
+  const auto end = std::max<BigInt::llong_t>(
+    std::min<BigInt::llong_t>(
+      args_value.size() > 2 ? args_value[2].to_long()
```

Comment:
I don't understand what the purpose of all these number conversions is. It would be much simpler to use `mp_integer` throughout.

Reply:
@peterschrammel To make that easier I added max and min functions to `mp_integer` in the first commit, can you have a look at that?